django/db/models/fields/__init__.py

```diff
@@ -772,6 +769,8 @@ def get_db_prep_lookup(self, lookup_type, value, connection,
             return [self.get_db_prep_value(value, connection=connection,
                                            prepared=prepared)]
         elif lookup_type in ('range', 'in'):
+            if not isinstance(value, (list, tuple)):
```

Comment:
I've totally reverted all the changes I made to the `Lookup` class, because what I was trying to do was just not going to work. I was trying to ensure that all expressions were always resolved, but I couldn't even make a guarantee that `resolve_expression` would be called on the `Lookup` class anyway. The existing code actually was wrapping each element correctly with the bilateral transformation.

I'm going to open up a separate cleanup ticket that will focus on always resolving expressions.